File: app/src/main/java/com/verbatoria/di/calendar/CalendarComponent.java
package com.verbatoria.di.calendar;

import com.verbatoria.ui.calendar.view.add.children.ChildActivity;
import com.verbatoria.ui.calendar.view.add.children.search.SearchChildrenActivity;
import com.verbatoria.ui.calendar.view.add.clients.ClientsActivity;
import com.verbatoria.ui.calendar.view.add.clients.search.SearchClientsActivity;
import com.verbatoria.ui.calendar.view.detail.EventDetailActivity;

import dagger.Subcomponent;

/**
 * Компонент Даггера для календаря
 *
 * @author nikitaremnev
 */
@Subcomponent(modules = {CalendarModule.class})
@CalendarScope
public interface CalendarComponent {

    void inject(ChildActivity childActivity);

    void inject(ClientsActivity clientActivity);

    void inject(EventDetailActivity eventDetailActivity);

    void inject(SearchClientsActivity searchClientsActivity);

    void inject(SearchChildrenActivity searchChildrenActivity);

}

File: app/src/main/java/com/verbatoria/di/dashboard/calendar/CalendarComponent.kt
package com.verbatoria.di.dashboard.calendar

import com.verbatoria.di.BaseInjector
import com.verbatoria.di.FragmentScope
import com.verbatoria.ui.dashboard.calendar.CalendarFragment
import dagger.Subcomponent

/**
 * @author n.remnev
 */

@Subcomponent(modules = [CalendarModule::class])
@FragmentScope
interface CalendarComponent : BaseInjector<CalendarFragment> {

    @Subcomponent.Builder
    interface Builder {

        fun build(): CalendarComponent

    }

}
